src/flashcart/ed64/ed64.c

```diff
@@ -175,42 +190,67 @@ static flashcart_err_t ed64_load_file (char *file_path, uint32_t rom_offset, uin
 }
 
 static flashcart_err_t ed64_load_save (char *save_path) {
-    void *address =  NULL;
-    ed64_save_type_t type = ed64_ll_get_save_type();
-
-    switch (type) {
-        case SAVE_TYPE_EEPROM_4K:
-        case SAVE_TYPE_EEPROM_16K:
-        case SAVE_TYPE_SRAM:
-        case SAVE_TYPE_SRAM_128K:
-        case SAVE_TYPE_FLASHRAM:
-            address = (void *) (SRAM_ADDRESS);
-            break;
-        case SAVE_TYPE_NONE:
-        default:
-            return FLASHCART_ERR_ARGS;
-    }
-
     FIL fil;
     UINT br;
 
     if (f_open(&fil, strip_sd_prefix(save_path), FA_READ) != FR_OK) {
-        return FLASHCART_ERR_LOAD;
+            f_close(&fil);
+            return FLASHCART_ERR_LOAD;
     }
 
-    size_t save_size = f_size(&fil);
+    size_t save_size = file_get_size(strip_sd_prefix(save_path));
+    uint8_t cartsave_data[save_size];
 
-    if (f_read(&fil, address, save_size, &br) != FR_OK) {
-        f_close(&fil);
-        return FLASHCART_ERR_LOAD;
+    if (f_read(&fil, cartsave_data, save_size, &br) != FR_OK) {
+            f_close(&fil);
+            return FLASHCART_ERR_LOAD;
     }
 
     if (f_close(&fil) != FR_OK) {
         return FLASHCART_ERR_LOAD;
     }
 
-    if (br != save_size) {
-        return FLASHCART_ERR_LOAD;
+    // everdrive doesnt care about the save type other than eeprom
+    // so we can just check the size
+
+    if (save_size == KiB(128)) { //sram 128 and flash
+        setFlashRAM(cartsave_data, save_size);
+    }
+
+    else if (save_size > KiB(2)) { //sram
+        setSRAM(cartsave_data, save_size);
+    }
+    
+    else if (save_size >= 512){ // eeprom
+        setEeprom(cartsave_data, save_size);
+    }
+
+    FIL lsp_fil;
+    UINT lsp_bw;
+
+    if (f_open(&lsp_fil, LAST_SAVE_FILE_PATH, FA_WRITE | FA_CREATE_ALWAYS) != FR_OK) {
+         return FLASHCART_ERR_LOAD;
+    }
+    if (f_write(&lsp_fil, strip_sd_prefix(save_path) , strlen(save_path), &lsp_bw) != FR_OK) {
+         f_close(&lsp_fil);
+         return FLASHCART_ERR_LOAD;
+    }
+
+    if (f_close(&lsp_fil) != FR_OK) {
+         return FLASHCART_ERR_LOAD;
+    }
+
+    FIL rsfil;
+
+    // this wries a 1 byte file as it only needs to exist to detect a reset
+
+    if (f_open(&rsfil, "/menu/RESET", FA_CREATE_ALWAYS ) != FR_OK) {
```

Comment:
To make this easier to maintain and change, use a define (as done with LAST_SAVE_FILE_PATH)